src/mrack/providers/openstack.py

```diff
@@ -420,21 +421,43 @@ async def create_server(self, req):
         if specs.get("network"):
             del specs["network"]
 
-        response = await self.nova.servers.create(server=specs)
+        error_attempts = 0
+        while True:
+            try:
+                response = await self.nova.servers.create(server=specs)
+            except ServerError as exc:
+                logger.debug(exc)
+                error_attempts += 1
+                if error_attempts > SERVER_ERROR_RETRY:
+                    raise ProvisioningError(
+                        f"{self.dsp_name}: Fail to create server", specs
+                    ) from exc
+                await asyncio.sleep(1)
```

Comment:
Should not we consider longer sleep time ? For example `self.poll_sleep` because this fix should behave better when openstack is overwhelmed and response from server is 500. Should not we wait a little ?

Reply:
What wait time would you propose?

Reply:
To respect timeout to provision all the systems it should be either smaller than that or to use `self.poll_sleep` or longer `self.poll_sleep_initial`. Does it make sense or it does not matter in this particular case?

Reply:
nova.servers.create just issues a creation of a server. I'm not sure if the timeout makes a big difference. Under load, the issuance might both fail or succeed the next second or it might also fail if we wait 10 mins. 

Maybe it could be reasonable to wait e.g. 10s assuming that there was a short spike in load which can be gone.

Reply:
One more thing about this (using self.poll_sleep) is that i wanted to get rid of "magic" numbers.

Reply:
Moved that to a constant as we discussed on IRC.